src/components/UnitPreferencesDialog.tsx

```diff
@@ -62,8 +63,16 @@ const UnitPreferencesDialog = ({ open, onClose }: Props) => {
     const handleToUnitChange = (e: SelectChangeEvent, unit: string) => {
         if (unit === 'temp') {
             dataActionDispatcher({ type: 'updateTempToUnit', unit: e.target.value as string });
+            setUnitPreferences({
```

Comment:
We can move this to `store/reducers.ts` and update them in `updateTempToUnit` and `updateDepthToUnit`.
It's a good practice to consolidate relevant logics/actions in one place. It makes refactoring easier.